File: C++/Hard/WrongAnswer/count-array-pairs-divisible-by-k_7.cpp
class Solution {
public:
    long long countPairs(vector<int>& nums, int k) {
        
        int n = nums.size();
        vector<int> cnt(k);
        long long res = 0;
        for (auto x : nums) {
            int y = x % k;
            if (y == 0) {
                res += cnt[0];
            } else {
                res += cnt[k - y];
            }
            cnt[y]++;
        }
        return res;
    }
};

File: Java/Hard/WrongAnswer/count-array-pairs-divisible-by-k_7.java
class Solution {
    public long countPairs(int[] nums, int k) {
        
        long res = 0;
        int[] cnt = new int[k];
        for (int i = 0; i < nums.length; i++) {
            int n = nums[i] % k;
            if (n < 0) {
                n += k;
            }
            res += cnt[(k - n) % k];
            cnt[n]++;
        }
        return res;
    }
}